src/com/dabomstew/pkrandom/BatchModeSettings.java

```diff
@@ -0,0 +1,125 @@
+package com.dabomstew.pkrandom;
+
+/*----------------------------------------------------------------------------*/
+/*--  BatchModeSettings.java - handles functionality related to batch mode. --*/
+/*--                                                                        --*/
+/*--  Part of "Universal Pokemon Randomizer ZX" by the UPR-ZX team          --*/
+/*--  Originally part of "Universal Pokemon Randomizer" by Dabomstew        --*/
+/*--  Pokemon and any associated names and the like are                     --*/
+/*--  trademark and (C) Nintendo 1996-2020.                                 --*/
+/*--                                                                        --*/
+/*--  The custom code written here is licensed under the terms of the GPL:  --*/
+/*--                                                                        --*/
+/*--  This program is free software: you can redistribute it and/or modify  --*/
+/*--  it under the terms of the GNU General Public License as published by  --*/
+/*--  the Free Software Foundation, either version 3 of the License, or     --*/
+/*--  (at your option) any later version.                                   --*/
+/*--                                                                        --*/
+/*--  This program is distributed in the hope that it will be useful,       --*/
+/*--  but WITHOUT ANY WARRANTY; without even the implied warranty of        --*/
+/*--  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          --*/
+/*--  GNU General Public License for more details.                          --*/
+/*--                                                                        --*/
+/*--  You should have received a copy of the GNU General Public License     --*/
+/*--  along with this program. If not, see <http://www.gnu.org/licenses/>.  --*/
+/*----------------------------------------------------------------------------*/
+
+import java.util.StringJoiner;
+
+public class BatchModeSettings implements Cloneable {
+    private Boolean batchModeEnabled;
+    private Boolean generateLogFile;
+    private Boolean autoAdvanceStartingIndex;
+    private Integer numberOfSeeds;
```

Comment:
I chose to use Boolean/Integer instead of their primitive counterparts so that I could use their toString() methods in the toString() method of BatchModeSettings. I felt like that read better than, for example, `sj.add("batchmode.enabled=" + Boolean.toString(batchModeEnabled));`